modules/aws/sonar-upgrader/python_upgrader/tests/test_main.py

```diff
@@ -300,7 +303,13 @@ def test_main_node_info_failure_with_stop_on_failure_false(
     # given
     setup_custom_args(args, [{"main": gw1}, {"main": gw2}], [{"main": hub1}], True, True, True, True, False)
     # fail if host is host1
-    collect_node_info_mock.side_effect = (lambda en: not en.get('dsf_node').get('host') == 'host1')
```

Comment:
Function collect_node_info is part of the tested code and should not be mocked. If you would have put this function in a separate file, e.g., sonar_utils.py, then you could mock it. The rule is, main_test.py tests the file main.py and only functions external to main.py should be mocked. (P0)

Reply:
What about `test_connection`, `run_remote_script`, `join_paths`, and `read_file_contents` which are all main functions and were all mocked in this file previously?

Reply:
All the functions you mentioned weren't in the main.py when the test code was written, you moved them to the main.py and didn't change the test code (in previous PRs). You have two options: Move the functions back outside of main.py, or change the test code so that functions of main.py are not mocked.